server/community.go

```diff
@@ -64,3 +66,62 @@ func (s *Server) postPRWelcomeMessage(ctx context.Context, pr *model.PullRequest
 
 	return nil
 }
+
+func (s *Server) assignGreeter(ctx context.Context, pr *model.PullRequest, repo *Repository) error {
+	// Only assign an greeter for non-member PRs
+	if s.IsOrgMember(pr.Username) {
+		return nil
+	}
+
+	// Is it configured to have a greeting team to handle the PR?
+	if repo.GreetingTeam == "" {
+		return nil
+	}
+
+	// get available members
+	greeterList, _, err := s.GithubClient.Teams.ListTeamMembersBySlug(ctx, s.Config.Org, repo.GreetingTeam, nil)
+	if err != nil {
+		mlog.Error("Error retrieving greeting team members", mlog.Err(err), mlog.String("GreetingTeam", repo.GreetingTeam))
+		return err
+	}
+
+	// Assign one of them as a reviewer
+	size := len(greeterList)
+	if size == 0 {
+		mlog.Warn("There were no members on the greeting team", mlog.String("GreetingTeam", repo.GreetingTeam))
+		return nil
+	}
+
+	greeter := greeterList[rand.Intn(size)]
+	greetingRequest := github.ReviewersRequest{
+		Reviewers: []string{*greeter.Login},
+	}
+
+	_, _, err = s.GithubClient.PullRequests.RequestReviewers(ctx, pr.RepoOwner, pr.RepoName, pr.Number, greetingRequest)
+	if err != nil {
+		mlog.Warn("Couldn't assing a member of the greeting team", mlog.Err(err), mlog.String("GreetingTeam", repo.GreetingTeam))
```

Comment:
Any reason not to return the error here? This seems like a failure, and we should just return the error with context and let the upper layer log it.